addons/web_editor/models/field_html_history_mixin.py

```diff
@@ -0,0 +1,104 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+
+from .patch_utils import apply_patch, generate_comparison, generate_patch
+
+
+class HtmlHistory(models.AbstractModel):
+    _name = "field.html.history.mixin"
+    _description = "Field html History"
+    _max_history_revisions = 300
+
+    history_revisions = fields.Json("History revisions Data")
+
+    @api.model
+    def _get_versioned_field(self):
+        """This method should be overriden
+
+        :return: List[string]: A list of name of the fields to be versioned
+        """
+        return []
+
+    def write(self, vals):
+        new_revisions = False
+        versioned_fields = self._get_versioned_field()
+        if any(x in versioned_fields for x in vals):
+            self.ensure_one()
+
+            history_revisions = (
+                self.history_revisions if self.history_revisions else {}
+            )
+
+            for field_name in versioned_fields:
+                new_content = vals[field_name] if field_name in vals else False
+
+                if field_name not in history_revisions:
+                    history_revisions[field_name] = []
+
+                if (
+                    new_content
+                    and self.id
+                    and isinstance(vals[field_name], str)
+                ):
+                    old_content = getattr(self, field_name)
+                    if (
+                        isinstance(old_content, str)
+                        and new_content != old_content
+                    ):
+                        new_revisions = True
+                        patch = generate_patch(new_content, old_content)
+
+                        history_revisions[field_name].insert(
+                            0,
```

Comment:
What is this `0` ? What does it do ? Is it like "inserting at index 0", to _prepend_ ?